components/detrack/app/detrack.app.ts

```diff
@@ -1,13 +1,36 @@
 import { defineApp } from "@pipedream/types";
+import { axios } from "@pipedream/platform";
+import {
+  CreateJobParams, HttpRequestParams, JobResponse,
+} from "../common/types";
 
 export default defineApp({
   type: "app",
   app: "detrack",
-  propDefinitions: {},
   methods: {
-    // this.$auth contains connected account data
-    authKeys() {
-      console.log(Object.keys(this.$auth));
+    _baseUrl(): string {
+      return "https://app.detrack.com/api/v2";
+    },
+    async _httpRequest({
+      $ = this,
+      endpoint,
+      ...args
+    }: HttpRequestParams): Promise<object> {
+      return axios($, {
+        url: this._baseUrl() + endpoint,
+        headers: {
+          "Content-Type": "application/json",
+          "X-API-KEY": this.$auth.api_key,
```

Comment:
Move this to a getter method

```suggestion
          "X-API-KEY": this._apiKey(),
```

Reply:
Applied this